test/e2e/test_auditlogs.go

```diff
@@ -34,73 +39,68 @@ import (
+func AuditLogsSourceWithTestImpl(t *testing.T) {
+	project := os.Getenv(lib.ProwProjectKey)
 
 	auditlogsName := helpers.AppendRandomString("auditlogs-e2e-test")
 	targetName := helpers.AppendRandomString(auditlogsName + "-target")
 	topicName := helpers.AppendRandomString(auditlogsName + "-topic")
 	resourceName := fmt.Sprintf("projects/%s/topics/%s", project, topicName)
 
-	client := Setup(t, true)
-	defer TearDown(client)
-
-	config := map[string]string{
-		"namespace":       client.Namespace,
-		"auditlogssource": auditlogsName,
-		"project":         project,
-		"serviceName":     serviceName,
-		"methodName":      methodName,
-		"resourceName":    resourceName,
-		"targetName":      targetName,
-		"targetUID":       uuid.New().String(),
-		"type":            v1alpha1.AuditLogEventType,
-		"source":          fmt.Sprintf("%s/projects/%s", serviceName, project),
-		"subject":         fmt.Sprintf("%s/%s", serviceName, resourceName),
-	}
-	for k, v := range packages {
-		config[k] = v
-	}
-	installer := NewInstaller(client.Dynamic, config,
-		EndToEndConfigYaml([]string{"auditlogs_test", "istio"})...)
-
-	// Create the resources for the test.
-	if err := installer.Do("create"); err != nil {
-		t.Errorf("failed to create, %s", err)
-		return
-	}
-
-	//Delete deferred.
-	defer func() {
-		if err := installer.Do("delete"); err != nil {
-			t.Errorf("failed to delete, %s", err)
-		}
-		// Just chill for tick.
-		time.Sleep(60 * time.Second)
-	}()
-
-	gvr := schema.GroupVersionResource{
-		Group:    "events.cloud.google.com",
-		Version:  "v1alpha1",
-		Resource: "auditlogssources",
-	}
-
-	jobGVR := schema.GroupVersionResource{
-		Group:    "batch",
-		Version:  "v1",
-		Resource: "jobs",
-	}
-
-	if err := client.WaitForResourceReady(client.Namespace, auditlogsName, gvr); err != nil {
+	client := lib.Setup(t, true)
+	defer lib.TearDown(client)
+
+	// Create a target Job to receive the events.
+	job := resources.AuditLogsTargetJob(targetName, []v1.EnvVar{{
+		Name:  "SERVICENAME",
+		Value: serviceName,
+	}, {
+		Name:  "METHODNAME",
+		Value: methodName,
+	}, {
+		Name:  "RESOURCENAME",
+		Value: resourceName,
+	}, {
+		Name:  "TYPE",
+		Value: v1alpha1.AuditLogEventType,
+	}, {
+		Name:  "SOURCE",
+		Value: fmt.Sprintf("%s/projects/%s", serviceName, project),
+	}, {
+		Name:  "SUBJECT",
+		Value: fmt.Sprintf("%s/%s", serviceName, resourceName),
+	}, {
+		Name:  "TIME",
+		Value: "360",
+	}})
+	client.CreateJobOrFail(job, lib.WithServiceForJob(targetName))
+
+	// Create the AuditLogsSource.
+	eventsAuditLogs := kngcptesting.NewAuditLogsSource(auditlogsName, client.Namespace,
+		kngcptesting.WithAuditLogsSourceServiceName(serviceName),
+		kngcptesting.WithAuditLogsSourceMethodName(methodName),
+		kngcptesting.WithAuditLogsSourceProject(project),
+		kngcptesting.WithAuditLogsSourceResourceName(resourceName),
+		kngcptesting.WithAuditLogsSourceSink(metav1.GroupVersionKind{
+			Version: "v1",
+			Kind:    "Service"}, targetName))
```

Comment:
might be cleaner to save the v1 Service GVK in a variable, as I guess it will be used in many places?

Reply:
Added it as a variable.